### ISSUE TITLE
Multiple GPU Issue

### ISSUE THREAD

**Issue by wolfviking0** [CESNET/GPUJPEG#32]
Hi @MartinPulec,

I am facing an issue with multiple gpu, I want to have two process using GPUJPEG each of them using one of my GPU.

Here my current setup:
```
OSX 10.13.6
There are 2 devices supporting CUDA:

Device #0: "GeForce GTX 1080 Ti"
  Compute capability: 6.1
  Total amount of global memory: 11534144 kB
  Total amount of constant memory: 64 kB
  Total amount of shared memory per block: 48 kB
  Total number of registers available per block: 65536
  Multiprocessors: 28

Device #1: "GeForce GTX 1080"
  Compute capability: 6.1
  Total amount of global memory: 8388416 kB
  Total amount of constant memory: 64 kB
  Total amount of shared memory per block: 48 kB
  Total number of registers available per block: 65536
  Multiprocessors: 20
CUDA driver version:   10.1
CUDA runtime version:  10.1
Using Device #0:       GeForce GTX 1080 Ti (c.c. 6.1)
```

As you can see your function gpujpeg_print_devices_info is working but if I call gpujpeg_init_device(0) and gpujpeg_init_device(1) into different process it's failed with this error 
```
[GPUJPEG] [Error]  src/gpujpeg_common.c (line 98): Cannot get number of CUDA devices: no CUDA-capable device is detected.
```

Also if I use only one process with GPU ID 1, it's seems to be running on 0.

Thanks

Tony

**Comment by MartinPulec** [comment]
Hi Tony,

I am not sure if I'll be able to reproduce the problem because I am not sure if am able to make a mac supporting CUDA with multiple GPUs. I've, however, tried to do so in Linux and it worked fine for me using UltraGrid:
```
xpulec@bunny:~$ nvidia-smi

| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      3054      C   bin/uv                                       117MiB |
|    1      3109      C   bin/uv                                       117MiB |
+-----------------------------------------------------------------------------+
```

These are just 2 independent processes using the GPU with GPUJPEG.

Would you be able to provide a working example code presenting the problem? Or at least instructions how to replicate (eg. timing of the processes etc.).

Thanks,
Martin

**Comment by wolfviking0** [comment]
Hi Martin,

Here my configuration, currently I am using Python (I create a wrapper using swig). So I am creating the process using the multiprocessing API. 
I will test with pycuda to see if It give me the same error when I try to initialize cuda from a subprocess.
I was able to use another approach, inside the main process python, I instantiate two GPUJPEG instance and then pass them to the process, now it seems to be working.

Unfortunately I still have one trouble, when I try to set the device_id to the GPU 1 it seems to be running on 0. 
In order to choose the device I did 'gpujpeg_init_device(device_id, flags)' it should be enough ? I don't have to specify something else ?
